### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by Zobamba**
Integrated Login/Generate OTP endpoints 

fp2

https://github.com/user-attachments/assets/9f8cf076-71c7-4dc2-a2d4-3df57aca8065

### CODE REVIEW

File: src/components/otp/otp.tsx [hunk 1]
```diff
@@ -29,32 +70,32 @@ export const OTP: React.FC<OTPProps> = ({ onClose }) => {
...
           >
             Cancel
           </Button>
           <Button
             type="submit"
             className="w-full bg-[#891C69] hover:bg-[#974D7B] focus:outline-none focus:ring-2 focus:ring-[#974D7B]"
+            onClick={handleOTPSubmit}
+            disabled={login.status === 'pending'}
           >
+            {login.status === 'pending' && (
```
Comment (oluwatunmiisheii) [inline src/components/otp/otp.tsx:96]: Add this functionality to the button component, that's why I showed you the example of the internal console

File: src/components/otp/otp.tsx [hunk 2]
```diff
@@ -7,16 +8,56 @@ import {
...
+    mutationFn: async ({
+      email,
+      password,
+      otp,
+    }: {
+      email: string
+      password: string
+      otp: string
+    }) =>
+      post(`/api/auth`, {
```
Comment (oluwatunmiisheii) [inline src/components/otp/otp.tsx:38]: before submitting, we need to validate the otp

File: src/components/forget-password/forget-password.tsx
Comment (oluwatunmiisheii) [inline src/components/forget-password/forget-password.tsx:1]: rename file and folder to forgot-password

**Review by oluwatunmiisheii — CHANGES_REQUESTED** [review]

File: src/components/forget-password/forget-password.tsx
Comment (Zobamba) [reply src/components/forget-password/forget-password.tsx:1]: Ok

**Review by oluwatunmiisheii — APPROVED** [review]
😎

**Pull request merged**